client/src/components/common/Modal.tsx

```diff
@@ -163,54 +177,69 @@ export const Modal: React.FC<{
   }
 
   return ReactDOM.createPortal(
-    <>
-      <Background onClick={onCloseWrapper} />
-      <div className={wrapper} data-cy="modal">
-        <Content size={size} className={className}>
-          <SpaceBetweenDiv
-            className={css`
-              position: sticky;
-              top: 0;
-              padding-top: 1rem;
-              align-items: center;
-              margin-bottom: 0.5rem;
-              background-color: var(--mi-lighten-background-color) !important;
-              padding: 1rem;
-              border-radius: var(--mi-border-radius-x) var(--mi-border-radius-x)
-                0 0;
-              padding-bottom: 0.5rem !important;
-              background-color: inherit;
-              border-bottom: solid 1px rgba(125, 125, 125, 0.3);
-              z-index: 12;
-
-              ${!title ? "justify-content: flex-end !important;" : ""}
-              button {
-                ${!title ? "margin-bottom: 0.2rem;" : ""}
-              }
-
-              h2 {
-                flex: 85%;
-                max-width: 85%;
-                margin-bottom: 0 !important;
-              }
-            `}
-          >
-            {title && <h2>{title}</h2>}
-
-            <Button
-              className={close}
-              size="compact"
-              startIcon={<FaTimes />}
-              onClick={onCloseWrapper}
-              aria-label="close"
-            ></Button>
-          </SpaceBetweenDiv>
-          <ChildrenWrapper className={contentClassName} noPadding={noPadding}>
-            {children}
-          </ChildrenWrapper>
-        </Content>
-      </div>
-    </>,
+		<FocusTrap
+			focusTrapOptions={{
+				initialFocus: ".".concat(close)
+			}}
+		>
+			<div
+				id={id || dialogIDFromTitle(title)}
+				role="dialog"
+				aria-describedby={ariaDescribedBy || undefined}
+				aria-labelledby={dialogLabelFromTitle(title)}
+				aria-modal="true"
+				onKeyDown={(event) => {
+					event.key === "Escape" && onCloseWrapper(event)
+				}}
+			>
+				<Background onClick={onCloseWrapper} />
+				<div className={wrapper} data-cy="modal">
```

Comment:
Thanks @simonv3 for the info about Cypress. I saw this `data-cy` and wasn't sure if it might be moved to the `div role=dialog` above it. I guess if the tests don't do much now, there's not much risk either way.